stdlib/test/bit/test_bit.mojo

```diff
@@ -19,95 +19,378 @@ from bit import (
+        bit_reverse(var3),
+        SIMD[int32_t, simd_width](-1, 0, -2147483648, 1073741824),
+    )
+
+    alias var4 = SIMD[int64_t, simd_width](-1, 0, 1, 2)
+    assert_equal(
+        bit_reverse(var4),
+        SIMD[int64_t, simd_width](
+            -1, 0, -9223372036854775808, 4611686018427387904
+        ),
+    )
+
+
+def test_byte_swap_simd():
+    alias simd_width = 4
+    alias int16_t = DType.int16
+    alias int32_t = DType.int32
+    alias int64_t = DType.int64
+
+    alias var2 = SIMD[int16_t, simd_width](-0x0123, 0x0000, 0x0102, 0x0201)
+    assert_equal(
+        byte_swap(var2),
+        SIMD[int16_t, simd_width](0xDDFE, 0x0000, 0x0201, 0x0102),
+    )
+
+    alias var3 = SIMD[int32_t, simd_width](
+        -0x01234567, 0x01234567, 0x56789ABC, 0x89ABCDEF
+    )
+    assert_equal(
+        byte_swap(var3),
+        SIMD[int32_t, simd_width](
+            0x99BADCFE, 0x67452301, 0xBC9A7856, 0xEFCDAB89
+        ),
+    )
+
+    alias var4 = SIMD[int64_t, simd_width](
+        -0x0123456789ABCDEF,
+        0x0123456789ABCDEF,
+        0x56789ABCDEF01234,
+        0x23456789ABCDEF01,
+    )
+    assert_equal(
+        byte_swap(var4),
+        SIMD[int64_t, simd_width](
+            0x1132547698BADCFE,
+            0xEFCDAB8967452301,
+            0x3412F0DEBC9A7856,
+            0x01EFCDAB89674523,
+        ),
+    )
+
+
+def test_pop_count_simd():
+    alias simd_width = 4
+    alias int8_t = DType.int8
+    alias int16_t = DType.int16
+    alias int32_t = DType.int32
+    alias int64_t = DType.int64
+
+    alias var1 = SIMD[int8_t, simd_width](-114, 0, 100, 2**6)
+    assert_equal(pop_count(var1), SIMD[int8_t, simd_width](4, 0, 3, 1))
+
+    alias var2 = SIMD[int16_t, simd_width](-11444, 0, 3000, 2**13)
+    assert_equal(pop_count(var2), SIMD[int16_t, simd_width](8, 0, 7, 1))
+
+    alias var3 = SIMD[int32_t, simd_width](-111444, 0, 30000, 2**29)
+    assert_equal(pop_count(var3), SIMD[int32_t, simd_width](22, 0, 7, 1))
+
+    # alias var4 = SIMD[int64_t, simd_width](-111444444, 0, 3000000, 2**59)
+    alias var4 = SIMD[int64_t, simd_width](
+        -111444444, 0, 3000000, 576460752303423488
+    )
+    assert_equal(pop_count(var4), SIMD[int64_t, simd_width](51, 0, 10, 1))
+
+
+def test_bit_not_simd():
+    alias simd_width = 4
+    alias int8_t = DType.int8
+    alias int16_t = DType.int16
+    alias int32_t = DType.int32
+    alias int64_t = DType.int64
+
+    alias var1 = SIMD[int8_t, simd_width](-114, 0, 100, 2**6)
+    assert_equal(bit_not(var1), SIMD[int8_t, simd_width](113, -1, -101, -65))
+
+    alias var2 = SIMD[int16_t, simd_width](-11444, 0, 3000, 2**13)
+    assert_equal(
+        bit_not(var2), SIMD[int16_t, simd_width](11443, -1, -3001, -8193)
+    )
+
+    alias var3 = SIMD[int32_t, simd_width](-111444, 0, 30000, 2**29)
+    assert_equal(
+        bit_not(var3), SIMD[int32_t, simd_width](111443, -1, -30001, -536870913)
+    )
+
+    # alias var4 = SIMD[int64_t, simd_width](-111444444, 0, 3000000, 2**59)
```

Comment:
here and below: Is this another manifestation of the bug you mentioned? If so, can you please link the ticket you open with a #TODO here and everywhere else changes are need to be made?

Reply:
yep will do!